src/hypertable.c

```diff
@@ -823,12 +823,16 @@ ts_hypertable_lock_tuple_simple(Oid table_relid)
 					 errmsg("hypertable \"%s\" has already been updated by another transaction",
 							get_rel_name(table_relid)),
 					 errhint("Retry the operation again")));
+			pg_unreachable();
+			return false;
 		case TM_BeingModified:
 			ereport(ERROR,
 					(errcode(ERRCODE_LOCK_NOT_AVAILABLE),
 					 errmsg("hypertable \"%s\" is being updated by another transaction",
 							get_rel_name(table_relid)),
 					 errhint("Retry the operation again")));
+			pg_unreachable();
+			return false;
```

Comment:
There are also `TM_Invisible` and default cases below which looks identical. Maybe move `pg_unreachable()` with `return false;` at the end of the function instead and call `break` here instead?